### PR TITLE
Defang the CT Timebomb

### PR DESCRIPTION
**Pull request by darkdh**
https://codereview.chromium.org/2495583002

fix https://github.com/brave/browser-laptop/issues/6106

Auditor: @diracdeltas, @bridiver

### CODE REVIEW

**Comment by bbondy** [comment]
This looks like a different patch than I rebased last night:

```
diff --git a/net/quic/crypto/proof_verifier_chromium.cc b/net/quic/crypto/proof_verifier_chromium.cc
index 096374a..7a0c3f3 100644
--- a/net/quic/crypto/proof_verifier_chromium.cc
+++ b/net/quic/crypto/proof_verifier_chromium.cc
@@ -345,6 +345,8 @@ int ProofVerifierChromium::Job::DoVerifyCertComplete(int result) {
     int ct_result = OK;
     if (verify_details_->ct_verify_result.cert_policy_compliance !=
             ct::CertPolicyCompliance::CERT_POLICY_COMPLIES_VIA_SCTS &&
+        verify_details_->ct_verify_result.cert_policy_compliance !=
+            ct::CertPolicyCompliance::CERT_POLICY_BUILD_NOT_TIMELY &&
         transport_security_state_->ShouldRequireCT(
             hostname_, cert_verify_result.verified_cert.get(),
             cert_verify_result.public_key_hashes)) {
diff --git a/net/socket/ssl_client_socket_impl.cc b/net/socket/ssl_client_socket_impl.cc
index d1edfbb..79aa8ee 100644
--- a/net/socket/ssl_client_socket_impl.cc
+++ b/net/socket/ssl_client_socket_impl.cc
@@ -1819,6 +1819,8 @@ int SSLClientSocketImpl::VerifyCT() {
 
   if (ct_verify_result_.cert_policy_compliance !=
           ct::CertPolicyCompliance::CERT_POLICY_COMPLIES_VIA_SCTS &&
+      ct_verify_result_.cert_policy_compliance !=
+          ct::CertPolicyCompliance::CERT_POLICY_BUILD_NOT_TIMELY &&
       transport_security_state_->ShouldRequireCT(
           host_and_port_.host(), server_cert_verify_result_.verified_cert.get(),
           server_cert_verify_result_.public_key_hashes)) {
diff --git a/net/spdy/spdy_session.cc b/net/spdy/spdy_session.cc
index 0e61326..78fb546 100644
--- a/net/spdy/spdy_session.cc
+++ b/net/spdy/spdy_session.cc
@@ -672,6 +672,8 @@ bool SpdySession::CanPool(TransportSecurityState* transport_security_state,
 
   if (ssl_info.ct_cert_policy_compliance !=
           ct::CertPolicyCompliance::CERT_POLICY_COMPLIES_VIA_SCTS &&
+      ssl_info.ct_cert_policy_compliance !=
+          ct::CertPolicyCompliance::CERT_POLICY_BUILD_NOT_TIMELY &&
       transport_security_state->ShouldRequireCT(
           new_hostname, ssl_info.cert.get(), ssl_info.public_key_hashes)) {
     return false;
```


Is only the one above needed or is this needed too?

**Comment by bbondy** [comment]
OK @darkdh explained to me it is just index precision changes for the difference, I'm thinking we should probably use the one we already built from though even know the index precision changes are better.

**Comment by bbondy** [comment]
or better yet, merge this one into c54 and remove the change from master when we merge it.

**Comment by bbondy** [comment]
++ I think we'll probably end up resetting to C54 so merging in.

**Comment by bridiver** [comment]
there are a lot of changes in here that shouldn't be there. I force pushed over it and it needs to be updated to only change those files and not change the diff for anything else

File: patches/master_patch.patch [hunk 1]
```diff
@@ -1,5 +1,5 @@
 diff --git a/build/config/chrome_build.gni b/build/config/chrome_build.gni
-index c649018..a3a5b95 100644
```
Comment (bridiver) [inline patches/master_patch.patch:2]: none of these changes should be here down to line 1619 and everything after 1659

File: patches/v8/filter.patch [hunk 2]
```diff
@@ -1,5 +1,5 @@
 diff --git a/BUILD.gn b/BUILD.gn
-index dcefe37..c234e79 100644
```
Comment (bridiver) [inline patches/v8/filter.patch:2]: there should not be any changes to this file

**Pull request merged**